### PR TITLE
Cloud Utilities Improvement 

### PR DESCRIPTION
**Pull request by vwxyzjn**
This PR improves the cloud workflow. See the changed readme `cloud/README.md`.

### CODE REVIEW

File: cleanrl_utils/submit_exp.py [hunk 1]
```diff
@@ -1,97 +1,70 @@
...
-    f.write(final_str)
+if args.build:
+    subprocess.run(
+        f"docker buildx build --push --cache-to type=local,mode=max,dest=docker_cache/cleanrl --cache-from type=local,src=docker_cache/cleanrl --platform {args.archs} -t {args.docker_tag} .",
+        shell=True,
+        check=True,
+    )
 
 if not args.wandb_key:
     args.wandb_key = requests.utils.get_netrc_auth("https://api.wandb.ai")[-1]
```
Comment (dosssman) [inline cleanrl_utils/submit_exp.py:54]: Regarding the section about the Wandb configuration when submitting experiments to the cloud, this is the relevant line.
I think the API what is returned by `request.utils.get_netrc_auth` is not as expected.

**Review by dosssman — APPROVED** [review]
## Regarding the documentation on of the cloud features

[Here](https://docs.cleanrl.dev/cloud/installation/), shouldn't the third line `cd cleanrl/cloud` be `cd cloud` instead ? Since we assume we are already in the cleanrl root folder.

Futhermore, a link to `terraform` installation could be useful.
I followed this one to setup terraform on the fresh machine I did the tests on: https://learn.hashicorp.com/tutorials/terraform/install-cli

## Regarding the script submissions:

Following the instructions [here](https://docs.cleanrl.dev/cloud/installation/), when trying to run any of the provided commands, I had to manually specify the `--wandb-key` to the scripts.

More specifically, if I just run the first command:
```bash
poetry run python -m cleanrl_utils.submit_exp \
    --docker-tag vwxyzjn/cleanrl:latest \
    --command "poetry run python cleanrl/ppo.py --gym-id CartPole-v1 --total-timesteps 100000 --track --capture-video" \
    --num-seed 1 \
    --num-vcpu 1 \
    --num-memory 2000 \
    --num-hours 48.0
```
 for example, I get the error:
```
$ poetry run python -m cleanrl_utils.submit_exp \>     --docker-tag vwxyzjn/cleanrl:latest \>     --command "poetry run python cleanrl/ppo.py --gym-id CartPole-v1 --total-timesteps 100000 --track --capture-video" \
>     --num-seed 1 \
>     --num-vcpu 1 \
>     --num-memory 2000 \
>     --num-hours 48.0
Traceback (most recent call last):
  File "/home/d055/anaconda3/envs/cleanrl-cloud/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/d055/anaconda3/envs/cleanrl-cloud/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/d055/random/rl/cleanrl/cleanrl_utils/submit_exp.py", line 54, in <module>
    args.wandb_key = requests.utils.get_netrc_auth("https://api.wandb.ai")[-1]
TypeError: 'NoneType' object is not subscriptable
```

For context, I only have the WANDB_API_KEY environment variable set, and the following command works as expected:
```bash
poetry run python -m cleanrl_utils.submit_exp \
    --docker-tag vwxyzjn/cleanrl:latest \
    --command "poetry run python cleanrl/ppo.py --gym-id CartPole-v1 --total-timesteps 100000 --track --capture-video" \
    --num-seed 1 \
    --num-vcpu 1 \
    --num-memory 2000 \
    --num-hours 48.0 \
    --wandb-key $WANDB_API_KEY
```

Is there any additional command that needs to be run to have the WANDB login credentials accessible through the `requests.utils.get_netrc_auth("https://api.wandb.ai")[-1]` module ?
(`python run wandb login` told me that I was already logged ...)

It might be also useful to have more specifics on how to handle the WANDB integration in the cloud documentation.

Beside that, everything is working smoothly. Great work :+1:

File: cleanrl_utils/submit_exp.py (hunk 1 above)
Comment (dosssman) [reply cleanrl_utils/submit_exp.py:54]: I think the check for the wandb key is not working as expected.

Comment (vwxyzjn) [reply cleanrl_utils/submit_exp.py:54]: Thanks for testing it out. Can you try `wandb login` after you remove the `WANDB_API_KEY` variable?

**Comment by vwxyzjn** [comment]
Thanks @dosssman for the review! I have updated the docs a bit more 

> Here, shouldn't the third line cd cleanrl/cloud be cd cloud instead?

This is fixed.

Could you give this PR a final look and I will merge it?

File: cleanrl_utils/submit_exp.py (hunk 1 above)
Comment (dosssman) [reply cleanrl_utils/submit_exp.py]: Explicitly logging in with the `wandb login` command and specifying the WANDB Api key seems to make it work as intended.

(Before, the logging-in was done implicitly when queuing up experiments using a simple `runs.sh`, so it was probably not adding the relevant authentication files to the .netrc file).

Comment (vwxyzjn) [reply cleanrl_utils/submit_exp.py]: Hmm, let's keep it as it is. We can't handle all the special cases.

**Review by dosssman — APPROVED** [review]
Looking good on my side.
Great job as always.

File: cleanrl_utils/submit_exp.py (hunk 1 above)
Comment (dosssman) [reply cleanrl_utils/submit_exp.py]: I concur.

**Pull request merged**